File: server/src/main/java/com/andrekj/ghostwriter/dto/BlogRequest.java
package com.andrekj.ghostwriter.dto;

import com.andrekj.ghostwriter.interfaces.BaseRequest;
import lombok.Data;

@Data
public class BlogRequest implements BaseRequest {
    private String contentType;
    private int wordCount;
    private String aimodel;
    private String topic;
    private String targetAudience;
    private String tone;
    private String expertiseLevel;
    private boolean seoFocus;
}

File: client/src/types/BlogRequest.ts
export default interface BlogRequestType {
    aimodel: {
        model: string;
        tooltip: string;
    };
    contentType: string;
    topic: string;
    targetAudience: string;
    tone: string;
    expertiseLevel: string;
    wordCount: number;
    seoFocus: boolean;
}